### PR TITLE
Enable C4267 warning

### PR DESCRIPTION
**Pull request by Vaisman**
* conversion from 'size_t' to 'type', by add static_cast
* extend type from int to int64_t

Tested:
* by build solution on Windows, Linux locally,
* run tests

### CODE REVIEW

**Comment by yuslepukhin** [comment]
@Vaisman Thanks for helping out. I will look at it. I have these changes in my private branch as well unpublished because of another PR pending which is not currently reviewed.
@siying @igorcanadi Do you have any objections to this and  https://github.com/facebook/rocksdb/pull/846 requests. I realize these might be too many changes but these two appear to be important for the correct functionality on Windows.

File: port/win/env_win.cc [hunk 1]
```diff
@@ -112,7 +112,8 @@ SSIZE_T pwrite(HANDLE hFile, const char* src, size_t numBytes,
 
   unsigned long bytesWritten = 0;
 
-  if (FALSE == WriteFile(hFile, src, numBytes, &bytesWritten, &overlapped)) {
+  if (FALSE == WriteFile(hFile, src, static_cast<DWORD>(numBytes), &bytesWritten,
```
Comment (yuslepukhin) [inline port/win/env_win.cc]: Please, add this before the call:
`assert(numBytes <= std::numeric_limits<DWORD>::max());`

File: port/win/env_win.cc [hunk 2]
```diff
@@ -954,7 +956,7 @@ class WinWritableFile : public WritableFile {
 
     DWORD bytesWritten = 0;
```
Comment (yuslepukhin) [inline port/win/env_win.cc]: Add ` assert(numBytes <= std::numeric_limits<DWORD>::max());` before the OS call

File: port/win/env_win.cc [hunk 3]
```diff
@@ -134,7 +135,8 @@ SSIZE_T pread(HANDLE hFile, char* src, size_t numBytes, uint64_t offset) {
 
   unsigned long bytesRead = 0;
 
-  if (FALSE == ReadFile(hFile, src, numBytes, &bytesRead, &overlapped)) {
```
Comment (yuslepukhin) [inline port/win/env_win.cc]: Add assert(numBytes <= std::numeric_limits<DWORD>::max()); before the OS call

File: db/db_compaction_test.cc [hunk 4]
```diff
@@ -160,7 +160,7 @@ const SstFileMetaData* PickFileRandomly(
       auto result = rand->Uniform(file_id);
       return &(level_meta.files[result]);
     }
-    file_id -= level_meta.files.size();
+    file_id -= static_cast<unsigned int>(level_meta.files.size());
```
Comment (yuslepukhin) [inline db/db_compaction_test.cc]: I feel that uint32_t is a shorter option that would preserve the size across the platforms

File: utilities/document/json_document_builder.cc [hunk 5]
```diff
@@ -38,7 +38,8 @@ bool JSONDocumentBuilder::WriteEndObject() {
 
 bool JSONDocumentBuilder::WriteKeyValue(const std::string& key,
                                         const JSONDocument& value) {
-  size_t bytesWritten = writer_->writeKey(key.c_str(), key.size());
```
Comment (yuslepukhin) [inline utilities/document/json_document_builder.cc]: Prepend with assert(key.size() <= std::numeric_limits<uint8_t>::max());

**Comment by yuslepukhin** [comment]
@Vaisman Please, run both db_test and individual tests. Recommend setting gflags /i <image> +vrf

**Comment by Vaisman** [comment]
@yuslepukhin Thanks for your comments,  I''l fix it.

**Comment by Vaisman** [comment]
Can somebody review?

**Comment by yuslepukhin** [comment]
@Vaisman @siying This looks good to me as far as I can see.

**Pull request merged**